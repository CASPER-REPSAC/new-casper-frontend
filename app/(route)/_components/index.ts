'use client';

export { default as Background } from './Background';
export { default as NoticeSection } from './NoticeSection';
export { default as TitleSection } from './TitleSection';
