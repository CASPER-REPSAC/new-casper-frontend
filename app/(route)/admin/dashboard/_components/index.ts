'use client';

export { default as FileLogTable } from './FileLogTable';
export { default as CommentLogTable } from './CommentLogTable';
export { default as PostLogTable } from './PostLogTable';
export { default as LoginLogTable } from './LoginLogTable';
