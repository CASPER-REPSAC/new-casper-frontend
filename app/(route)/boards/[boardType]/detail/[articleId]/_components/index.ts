export { default as DetailComment } from './DetailComment';
export { default as AuthorSection } from './AuthorSection';
export { default as ContentSection } from './ContentSection';
export { default as ButtonSection } from './ButtonSection';
export { default as CommentEditorSection } from './CommentEditorSection';
