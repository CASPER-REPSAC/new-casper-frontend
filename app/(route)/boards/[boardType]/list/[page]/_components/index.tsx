export { default as BoardHeader } from './BoardHeader';
export { default as BoardFooter } from './BoardFooter';
export { default as BoardBody } from './BoardBody';
export { default as PostLink } from './PostLink';
