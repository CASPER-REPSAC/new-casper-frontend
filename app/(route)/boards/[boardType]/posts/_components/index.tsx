'use client';

export { default as EditorSection } from './EditorSection';
export { default as TitleSection } from './TitleSection';
export { default as WriteButtonSection } from './WriteButtonSection';
export { default as BoardTypeSelecSection } from './BoardTypeSelecSection';
export { default as PostFormProvider } from './PostFormProvider';
export { default as OptionSection } from './OptionSection';
