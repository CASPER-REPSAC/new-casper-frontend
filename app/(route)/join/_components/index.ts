'use client';

export { default as JoinFunnel } from './JoinFunnel';
export { default as PageWrapper } from './PageWrapper';
