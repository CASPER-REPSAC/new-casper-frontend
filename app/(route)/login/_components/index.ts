'use client';

export { default as Logo } from './Logo';
export { default as LoginForm } from './LoginForm';
export { default as LoginMoreLinks } from './LoginMoreLinks';
