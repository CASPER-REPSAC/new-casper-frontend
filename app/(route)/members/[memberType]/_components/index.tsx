'use client';

export { default as DetailMemberCard } from './DetailMemberCard';
export { default as MemberCard } from './MemberCard';
export { default as PageTemplate } from './PageTemplate';
