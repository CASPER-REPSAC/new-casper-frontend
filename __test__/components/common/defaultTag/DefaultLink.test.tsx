import { DefaultLink } from '@app/_components/common';
import { render, screen } from '@testing-library/react';

describe('<DefaultLink />', () => {
  test('snapshot test', () => {
    const { container } = render(
      <DefaultLink href="https://www.test.com">Link</DefaultLink>,
    );

    expect(container).toMatchSnapshot();
  });

  test('href 속성을 올바르게 가진다.', () => {
    const href = 'https://www.test.com';
    render(<DefaultLink href={href}>Link</DefaultLink>);

    const link = screen.getByRole('link');

    expect(link).toHaveAttribute('href', href);
  });
});
