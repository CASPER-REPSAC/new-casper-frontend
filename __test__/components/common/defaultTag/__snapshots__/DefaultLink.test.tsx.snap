// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultLink /> snapshot test 1`] = `
<div>
  <a
    class="block text-center btn px-4 py-2 text-base  undefined"
    href="https://www.test.com"
  >
    Link
  </a>
</div>
`;
